homeassistant/components/ecovacs/entity.py

```diff
@@ -150,6 +150,11 @@ def __init__(self, device: VacBot) -> None:
 
         self._event_listeners: list[EventListener] = []
 
+    @property
+    def available(self) -> bool:
+        """Return True if the entity is available."""
+        return super().available and self.state is not None
```

Comment:
Why the state is not None?

Reply:
as soon as we have received a valid state from the bot, the entity become available. This will prevent an supposedly available entity, but with a unknown state during startup of HA while waiting for connection to the bot